client_src/js/fileManagerRunner.js

```diff
@@ -56,7 +56,7 @@ class Manager {
               <div class="col-lg-10">
                 <div class="d-flex w-100 justify-content-between">
                   <h5 class="mb-1">
-                    <span id="printerName-${printer._id}">
+                    <span class="badge badge-primary" id="printerName-${printer._id}">
```

Comment:
Can this be secondary rather than primary, to fit with the other title's in the UI? 

Blue sticks out like a saw thumb in my opinion.

Reply:
Oh same for the title as well

Reply:
I find the secondary color quite weak in all the grays already present in the printer list. Secondly, there's little to nothing visually linking the selected printer to the shown file list except similar text (which isnt the same size and not the same transparency).
That's why I made it blue, for me it doesnt stand out as the blue is quite dim.

Try to look at the text I marked and see the problem with `badge-secondary`
![image](https://user-images.githubusercontent.com/6005355/124505482-e1b78d80-ddc9-11eb-8e52-738e722d6758.png)

Reply:
Oh and another point: the list hover color is not consistent making it a bit uncomfortable for me. Plus the fact that the cursor constantly changes, Im a bit on the edge of my seat there

Reply:
I agree the background now looks out of place on selected. That's originally why I had it as a text field with transparent background and not a badge/button. Changing that style seems a bit much now I've seen it in the screenshot, Maybe just revert to making both texts the same as the others on the file navigation bar? What do you think? 

I still think the "Files: # / Folders: #" text would go nicely under the "Operational" badge, did you try that at all? So would be 
"Operational"
"Files: #"
Folder #" 
and would utilise that space there a bit more rather than just having the printer icon. 

Personally regarding the colours I'd not colour the texts background as it's a stark difference to every other card around the interface. Maybe change the selection colour of the card? They alternate between grey and darker grey so a colour that stands out more may be better suited there. 

I've never had any complaints over readability there so maybe I'm just a bit blind sided by the "This is obviously the selected printers card list" so feel free to whack my out of that. Screenshots certainly help aha.

Reply:
Regarding the loading I've never personally noticed the long load times, even with searching which pulls the complete file list before filtering (this would still be needed as the search is recursive, so maybe a server side function for that). I wouldn't personally say pagination is required but having it just load the current directory rather than the whole structure is needed if there is indeed slow loading times. Would you say that's relevant to this PR? I'm leaning against not but 100% something that needs to be added.

Reply:
History needs the same as well

Reply:
About pagination: I dont mind deferring that. Would be nice to be able to benchmark such calls with the runOnce functionality I had in mind on the other `profiles.find`/task scheduler PR.

About the files/folders, I didnt realize you opted for the left: I like it. Will apply on this PR later.

About the printer selection, I would make the selected printer less dark and make the hover the same or nearly the same color (now its quite a difference).

About the current printer, there's also another option. We can make the other text on the file list title bar a different text to show those are "details" but the printer name is not/important. F.e. using <small> on the storage details or <strong> on the printer name. I do hope you agree we need something more than the same font and size. The problem is printer names can be long which means a big text shows up on small space.

About graycolors, I'd look for flattening the grays used a bit. Look at github dark mode for what I envision. It has a blue navbar and dark background. The rest is borders and some panel background gray. In total about 3/4 grays max. OF has about 6. I'd propose to bring that back to 3/4.

Sidenote: changing the selected printer or clicking a subfolder shifts the page. Hope to fix that. Annoys me a bit.

Reply:
> About pagination: I dont mind deferring that. Would be nice to be able to benchmark such calls with the runOnce functionality I had in mind on the other `profiles.find`/task scheduler PR.
> 
> About the files/folders, I didnt realize you opted for the left: I like it. Will apply on this PR later.
> 
> About the printer selection, I would make the selected printer less dark and make the hover the same or nearly the same color (now its quite a difference).
> 
> About the current printer, there's also another option. We can make the other text on the file list title bar a different text to show those are "details" but the printer name is not/important. F.e. using on the storage details or ** on the printer name. I do hope you agree we need something more than the same font and size. The problem is printer names can be long which means a big text shows up on small space.**
> 
> ** About graycolors, I'd look for flattening the grays used a bit. Look at github dark mode for what I envision. It has a blue navbar and dark background. The rest is borders and some panel background gray. In total about 3/4 grays max. OF has about 6. I'd propose to bring that back to 3/4. Sidenote: changing the selected printer or clicking a subfolder shifts the page. Hope to fix that. Annoys me a bit.**

Was on discord, no drama if you missed it regarding the left. 

I'm happy to see what you come up with regarding what you said. I'm in agreement it needs something more totally. 

Yeah the printer names is something I've struggled with quite a bit, I personally think truncating them anything more than an IP+Port length would suffice. Just going off mostly what I've seen of my users they like to call their printers cartoon names or something similar, but I'd gather larger farms would more go for a server naming convention so I've always been reserved to cut off the last set of letters as that's usually where the identifying part is.

Reply:
What about setting an avatar for each printer and/or using the first letter + a random color as a identifier if no avatar is set?
The idea is here, taken from Azure DevOps (or microsoft single sign on)
![image](https://user-images.githubusercontent.com/6005355/124566512-685c8100-de43-11eb-99c7-8ec972e2032e.png)

Discord does this brilliantly:
![image](https://user-images.githubusercontent.com/6005355/124566314-3a773c80-de43-11eb-8ffd-4099530c26db.png)

I guess we have much more creative room than you might think!

Reply:
My only issue with that is it's build around names which are generally two strings added togather so you get two sets of first letters. A farm using location codes wouldn't fit very well into that as they'd probably have rack-1-2 or colour designations. I also don't think it covers the character naming side as just for example, Micky/Mini would look the same. One more thing! IP's wouldn't work as you'd get "H" or "1" normally. 

I do agree there's more creative room here than is currently displayed in my early attempts. xD

Reply:
The thing is if we offer a way to organize, users will most probably use it and sanitize their printer names to fit. We just have to choose and take stacking/grouping/location placement into account.

Oh and Minnie/Miny => Mi#1 Mi#2

Reply:
> The thing is if we offer a way to organize, users will most probably use it and sanitize their printer names to fit. We just have to choose and take stacking/grouping/location placement into account.
> 
> Oh and Minnie/Miny => Mi#1 Mi#2

As someone who like's their custom naming scheme I would be annoyed at Mi#1 Mi#2 baha. Is Mini 1 or Micky 2, requires me to remember an order. It's something we need to discuss going forward anyway, and a user choice for what's displayed is always a good idea. It's worth a discussion maybe? then an issue when it's been worked out as it's a bit disjointed around the views currently.

Reply:
https://github.com/OctoFarm/OctoFarm/discussions/677